lib/pf/node.pm

```diff
@@ -1327,6 +1331,32 @@ sub _cleanup_attributes {
     $info->{'status'} = _cleanup_status_value($info->{'status'});
 }
 
+=item node_extended_data
+
+Get the node extended data
+
+=cut
+
+sub node_extended_data {
+    my ($mac) = @_;
+    my @namespaces = qw(mse);
+    if (@{$Config{advanced}{extended_profile_filter_namespaces}}) {
+        my %hash;
+        my $redis = pf::Redis->new(server => 'localhost:6379');
+        my $json = JSON::MaybeXS->new;
+        foreach my $namespace (@namespaces) {
+            my ($data) = $redis->get("extended:${namespace}:${mac}");
```

Comment:
Well example, I have a JSON blob in a key that contains the status of the mac in different MDM providers:
{
'MobileIron' : 0,
'OPSWAT' : 1,
'ZammitDM' : 0
}

Now if I want to change OPSWAT to 0, I have to fetch the redis key, decode the JSON blob, modity the hash, re-encode it and re-set it in the redis key. If in the meanwhile another operation is being done on that key to change 'ZammitDM' to 1, then it will be lost when I write the operation for OPSWAT.

If it would be a redis hash, I'd simply do `hset extended:whatever:00:11:22:33:44:55 OPSWAT 1` without risking to impact other keys in the hash.

Then in this piece of code doing HGETALL instead of decoding the JSON piece will give the whole hash.

The keys in the hash don't support complex structures so this is why I suggesting having both of them.